sbtc-core/src/signer/coordinator/mod.rs

```diff
@@ -25,8 +27,41 @@ pub struct PublicKeys {
 /// TODO: Define the Message types for Tx Signning Round
 /// https://github.com/stacks-network/sbtc/issues/43
 
+/// An sBTC transaction needing to be processed by the coordinator
+/// TODO: replace with the core library's SBTCTransaction
+/// This could be a BTC transaction or a STX transaction
+/// depending on https://github.com/Trust-Machines/stacks-sbtc/pull/595
+pub enum SBTCTransaction {
+    /// A commit Bitcoin transaction
+    Commit(TaprootSpendInfo, BitcoinTransaction),
+    /// A withdrawal Stacks transaction
+    Withdawal(StacksTransaction),
+}
+
+/// Revealer trait for revealing BTC commit transactions
+pub trait Revealer {
+    /// Retrieve pending Commit transactions information
+    fn pending_commit_transactions(
+        &self,
+    ) -> SBTCResult<Vec<(TaprootSpendInfo, BitcoinTransaction)>>;
+    /// Validate the given commit transaction
```

Comment:
This implies the state lives in the Revealer. I wonder if it would be better if we managed state somewhere else in a single place to ensure database consistency. The Revealer trait would only revolve around the functionality to validate commit transactions and reveal transactions. The process of interacting with a reveal server (or however we'll implement it) could be broken out to a separate entity.